src/nvim/if_cscope.c

```diff
@@ -1715,9 +1715,13 @@ static int cs_read_prompt(size_t i)
       }
     }
 
-    for (size_t n = 0; n < strlen(CSCOPE_PROMPT); ++n) {
-      if (n > 0)
+    for (size_t n = 0; n < strlen(CSCOPE_PROMPT); n++) {
+      if (ferror(csinfo[i].fr_fp) && errno == EINTR) {
```

Comment:
manpage for `ferror()` mentions that it does not modify `errno`, but I don't like having to check this. Specially if the code is changed later on. 

Actually, it isn't clear what function this `errno` check is associated with. It should be saved immediately after the function call.

```
int e = errno;
...
if (ferror(csinfo[i].fr_fp) && e == EINTR) {

```